src/hooks/useLoginProfile.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  exchangeCodeAsync,
+  makeRedirectUri,
+  refreshAsync,
+  useAuthRequest
+} from 'expo-auth-session';
+import { useEffect, useState } from 'react';
+
+import * as appJson from '../../app.json';
+import { addToStore, readFromStore } from '../helpers';
+
+const PROFILE_ACCESS_TOKEN = 'profileAccessToken';
+
+type TProfile = {
+  serverUrl: string;
+  clientId: string;
+  clientSecret: string;
+  scopes: string[];
+};
+
+export const useLoginProfile = (profile: TProfile) => {
+  const { clientId, clientSecret, scopes, serverUrl } = profile;
+
+  const keycloakDiscovery = {
+    authorizationEndpoint: serverUrl + '/auth',
+    tokenEndpoint: serverUrl + '/token',
+    revocationEndpoint: serverUrl + '/revoke'
+  };
+
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const redirectUri = makeRedirectUri({
+    scheme: appJson.expo.scheme,
+    path: 'redirect'
+  });
+
+  const [request, response, promptAsync] = useAuthRequest(
+    {
+      clientId,
+      redirectUri,
+      scopes,
+      usePKCE: false
+    },
+    keycloakDiscovery
+  );
+
+  useEffect(() => {
+    const handleAuthResponse = async () => {
+      if (response?.type === 'success') {
+        const { code } = response.params;
+        await exchangeToken(code);
+      }
+    };
+
+    handleAuthResponse();
+  }, [response]);
+
+  const exchangeToken = async (code: string) => {
+    try {
+      setLoading(true);
+      const tokenResult = await exchangeCodeAsync(
+        {
+          code,
+          clientId,
+          clientSecret,
+          redirectUri,
+          extraParams: {
+            grant_type: 'authorization_code'
+          }
+        },
+        keycloakDiscovery
+      );
+
+      setIsLoggedIn(true);
+      await addToStore(PROFILE_ACCESS_TOKEN, JSON.stringify(tokenResult));
```

Comment:
shouln't we use the `SecureStore` with `SecureStore.setItemAsync(...)` for the token?

Reply:
Yeah, you're right. I'll update it with `SecureStore`. 👍

Reply:
getting this now

> Value being stored in SecureStore is larger than 2048 bytes and it may not be stored successfully. In a future SDK version, this call may throw an error.

Reply:
We are currently using the `issueAt`, `expiresIn` and `refreshToken` values from the `tokenResult` object we saved in `SecureStore`. Different keys can be created for these 3 values and saved differently. We can also adapt the `token` value, which I think we will need in the future.

Reply:
Only `refreshToken` needs to be put in `SecureStore`, right? The others can be in `AsyncStorage`. What do you think is the effort to refactor this?

Reply:
I developed this code a long time ago. Therefore, I must first carefully analyse the changes that need to be made. I estimate it will take about two hours.